drivers/overlay/composefs_supported.go

```diff
@@ -80,6 +80,46 @@ func getErofsBlob(dataDir string) string {
 	return filepath.Join(dataDir, "erofs.blob")
 }
 
+type composefsAdditionalData struct {
+	NoACL bool `json:"noACL"`
+}
+
+func getComposefsAdditionalDataFile(dataDir string) string {
+	return filepath.Join(dataDir, "composefs.additional-data")
```

Comment:
Not a blocker but I think this would be cleaner as an xattr on the composefs data instead.

Also AIUI we only need this because we're not using `mount.composefs` which has https://github.com/containers/composefs/blob/ac729b579dd332938c6ad43ba83b2b896631dad4/libcomposefs/lcfs-mount.c#L404 right?

I'm a bit confused why we're not using `mount.composefs`.

Reply:
I was also unsure whether using a file or a xattr and I am fine to change it.

I've preferred a file as a xattr is not covered by fs-verity if we ever need it and it would be a problem for rootless on tmpfs (so with `--transient-store) since `user.` is not supported

Reply:
> I'm a bit confused why we're not using `mount.composefs`.

because `mount.composefs` uses a single erofs mount with multiple data layers under an overlay mount.
In c/storage we are using multiple erofs (one for layer) above multiple data layers under the same overlay mount that includes also the writeable upper layer where necessary.

The model in c/storage doesn't work well yet with multiple layers since the whiteouts in each data layer conflict with other layers, but once the "data only" overlay patches land upstream, we don't need these whiteouts anymore so the data paths are "append only" instead of hiding files below them.  And we need to decide on the prefix to use to avoid polluting the root directory, e.g. `.cfs/ab/123456...` instead of using `ab/123456...` as we do now.

Reply:
> because mount.composefs uses a single erofs mount with multiple data layers under an overlay mount.
In c/storage we are using multiple erofs (one for layer) above multiple data layers under the same overlay mount that includes also the writeable upper layer where necessary.

Right, OK.  Hmm.  

> . And we need to decide on the prefix to use to avoid polluting the root directory, e.g. .cfs/ab/123456... instead of using ab/123456... as we do now.

Yeah, agree on this.  Would definitely be cleaner to have one whiteout for `.cfs`.

Reply:
> I've preferred a file as a xattr is not covered by fs-verity if we ever need it 

But this state is locally computed; if we enabled fs-verity, what key would sign it?  There is I guess value in having it be immutable post-creation though.

Anyways backing up a second...maybe what would work a bit better is something like `mount.composefs --inspect /path/to/cfs.img` and that exports JSON that we can read even if we're doing the mounting on our own?  Since we already have this "has acl" flags in the composefs image right?

> and it would be a problem for rootless on tmpfs (so with --transient-store) since user.` is not supported

Ah yes :cry:  ostree uses `user.` xattrs in some modes and the tmpfs thing has been a tremendous pain.

Reply:
No need for that. The lcfs_erofs_header_s struct and the flags for wheter an ACL is in the image is public and ABI stable, we can just read it in go.